### PR TITLE
Feat: delete profilepicture button

### PR DESCRIPTION
**Pull request by Clemensito**
# Summary
Added a new "delete" for the profile picture funcitonnality.

# What
The button updates the databases fields for user and account collection to delete the user's profilepicture.

- UI
  - New button "delete" next to the edit profile picture 
- Documentation
  - Kdoc comments added/updated
- Tests
  - Added unit and integration tests for firebase and inMemory methods 
  - Added unit and integration tests for modelView and UI to ensure proper functionning

# Why
Needed for UX, and overall profile management

# Screenshots / Videos¨
Button :
image

After pressing it: 
image

# Checklist
- [x] Tests: added or updated (unit/instrumented/automation) as appropriate
- [x] Documentation: README/usage/docs updated where relevant
- [x] Manual verification: app run and basic flows checked (list steps in "What")

### CODE REVIEW

File: app/src/main/java/com/android/ootd/model/account/AccountRepositoryFirestore.kt [hunk 1]
```diff
@@ -351,6 +351,20 @@ class AccountRepositoryFirestore(private val db: FirebaseFirestore) : AccountRep
     }
   }
 
+  override suspend fun deleteProfilePicture(userID: String) {
+    try {
+      if (userID.isBlank()) throw IllegalArgumentException("Invalid userId")
```
Comment (j-meric) [inline app/src/main/java/com/android/ootd/model/account/AccountRepositoryFirestore.kt]: **Nitpick** As mentioned below, by adding an additional exception case you make the error management more concise and avoid repetition.

File: app/src/main/java/com/android/ootd/model/user/UserRepositoryFirestore.kt [hunk 2]
```diff
@@ -187,6 +187,20 @@ class UserRepositoryFirestore(private val db: FirebaseFirestore) : UserRepositor
     }
   }
 
+  override suspend fun deleteProfilePicture(userID: String) {
+    if (userID.isBlank()) throw IllegalArgumentException("UserID cannot be blank")
```
Comment (j-meric) [inline app/src/main/java/com/android/ootd/model/user/UserRepositoryFirestore.kt]: **Nitpick**: Same as above, add additional exception to increase robustness of the code.

File: app/src/main/java/com/android/ootd/model/user/UserRepository.kt [hunk 3]
```diff
@@ -75,4 +75,15 @@ interface UserRepository {
    * @throws Exception for other Firestore-related errors
    */
   suspend fun deleteUser(userID: String)
+
```
Comment (j-meric) [inline app/src/main/java/com/android/ootd/model/user/UserRepository.kt]: **Question** Why do you have the function deleteProfilePicture in the UserRepository? Even if at a later point in time we add the attribute profile picture to the user, it still makes more sense to adapt deleteProfilePicture to delete it from the attribute user from accountRepository for a clearer logic flow and avoiding duplication.

File: app/src/main/java/com/android/ootd/ui/account/AccountScreen.kt [hunk 4]
```diff
@@ -224,19 +225,38 @@ private fun AccountScreenContent(
...
+                  shape = CircleShape,
+                  colors = ButtonDefaults.buttonColors(containerColor = colors.primary),
+                  modifier = Modifier.testTag(UiTestTags.TAG_ACCOUNT_EDIT)) {
+                    Text(text = "Edit", color = colors.onPrimary, style = typography.titleLarge)
+                  }
+
+              Spacer(modifier = Modifier.width(16.dp))
+
+              // Delete button - only show if user has a profile picture
+              if (avatarUri.isNotBlank()) {
```
Comment (j-meric) [inline app/src/main/java/com/android/ootd/ui/account/AccountScreen.kt]: **Praise: ** Great idea of only showing delete button when there is a profile picture!

**Review by j-meric — CHANGES_REQUESTED** [review]
## PR Checklist Review

### Functional
- **Manual verification:** Key flows and edge cases verified — Great.
- **Automated tests:** Relevant unit/integration tests added or updated — Excellent (~95% coverage).
- **CI:** All checks pass.

### Code Quality
- **Readability:** Clear names and simple logic; avoid duplicated expressions (see Exception handling note).
- **Modularity:** Small, single-purpose functions/components — OK.
- **Style:** Linting and formatting OK.

### Security & Data
- **Validation/authorization:** Inputs validated and access checks present — OK.
- **Sensitive data:** Stored/transmitted securely — OK.

### Commits & PR Hygiene
- **Commit messages:** According to guideline — Great.
- **PR description:** According to template — Good.
- **Scope:** Changes limited to the feature/bug — Good, except for the modifications in the user repository, which would be nice to mention.

### Observability & Reliability
- **Error handling:** Graceful failures and meaningful logs — OK.

---

## Issues to Address
1. **Sonar issues:** Run Sonar, review reported findings, and fix any valid issues (or justify false positives).
2. **Exception handling duplication:** Add a specific exception case for a blank username instead of reusing the same expression three times; replace repeated expressions with the new case to improve clarity.
3. **Repository responsibility/design question:** Why is deleteProfilePicture in UserRepository? Consider moving or adapting it to operate on the user attribute in AccountRepository (or refactor/delegate) to clarify responsibilities and avoid duplication.

## Suggested UX Improvements 
- Make the delete button red to indicate destructive action.
- Change the edit button text to “Upload profile picture” when no picture exists.

Nice work — deleting the profile picture is a valuable addition to the account that enhances user experience and  tests, manual verification, docs, and commits are strong. Address the Sonar/design nitpicks and the small exception-handling cleanup.

File: app/src/androidTest/java/com/android/ootd/utils/AccountFirestoreTest.kt [hunk 5]
```diff
@@ -42,7 +42,7 @@ open class AccountFirestoreTest : FirestoreTest() {
             username = "bob_builder",
             birthday = "1992-05-15",
             googleAccountEmail = "bob@example.com",
-            profilePicture = "",
+            profilePicture = "bob.profilePic",
             friendUids = emptyList())
     dummyAccount =
         Account(
```
[the remark this answers is not in the document]
Comment (Clemensito) [reply app/src/androidTest/java/com/android/ootd/utils/AccountFirestoreTest.kt]: Thanks ! :)

File: app/src/main/java/com/android/ootd/model/user/UserRepository.kt (hunk 3 above)
Comment (Clemensito) [reply app/src/main/java/com/android/ootd/model/user/UserRepository.kt]: I agree but since we have 2 separate classes it made sense to add one method per class. But that is one thing we will need to refactor in a future PR

File: app/src/main/java/com/android/ootd/model/account/AccountRepositoryFirestore.kt (hunk 1 above)
Comment (Clemensito) [reply app/src/main/java/com/android/ootd/model/account/AccountRepositoryFirestore.kt]: Alright ! I'll do that

File: app/src/main/java/com/android/ootd/ui/account/AccountScreen.kt [hunk 6]
```diff
@@ -224,19 +225,38 @@ private fun AccountScreenContent(
...
+                    onClick = {
+                      accountViewModel.deleteProfilePicture()
+                      Toast.makeText(context, "Profile picture removed", Toast.LENGTH_SHORT).show()
+                    },
+                    shape = CircleShape,
+                    colors = ButtonDefaults.buttonColors(containerColor = colors.tertiary),
+                    modifier = Modifier.testTag(UiTestTags.TAG_ACCOUNT_DELETE)) {
+                      Text(text = "Delete", color = colors.onError, style = typography.titleLarge)
+                    }
               }
```
[the remark this answers is not in the document]
Comment (Clemensito) [reply app/src/main/java/com/android/ootd/ui/account/AccountScreen.kt]: Yep good idea
